tests/unit/model/projections/changed-attrs-test.js

```diff
@@ -88,4 +88,191 @@ module('unit/model/projections/changed-attrs', function(hooks) {
+      class TestAdapter {
+        static create() {
+          return new TestAdapter(...arguments);
+        }
+        updateRecord() {
+          return Promise.resolve();
+        }
+      }
+    );
+    this.owner.register(
+      'service:m3-schema',
+      class TestSchema extends DefaultSchema {
+        includesModel() {
+          return true;
+        }
+
+        computeNestedModel(key, value) {
+          assert.ok(true, 'Called computed nested model');
+          let attributesType;
+          try {
+            attributesType = value && value.$type;
+          } catch (e) {
+            assert.ok(false, "We don't pass Megamorphic Models to computeNestedModel");
+          }
+
+          if (key !== 'results' && typeof attributesType !== 'string') {
+            // object without a known type
+            return undefined;
+          }
+
+          return {
+            id: key,
+            type: attributesType,
+            attributes: value,
+          };
+        }
+
+        computeBaseModelName(modelName) {
+          return ['com.bookstore.projected-book', 'com.bookstore.excerpt-book'].includes(modelName)
+            ? 'com.bookstore.book'
+            : null;
+        }
+      }
+    );
+
+    this.store.push({
+      data: [
+        {
+          id: 'urn:book:1',
+          type: 'com.bookstore.Book',
+          attributes: {
+            locations: [
+              {
+                country: 'US',
+                geographicArea: 'California',
+                city: 'San Francisco',
+                postalCode: '94110',
+                description: 'Club house',
+                $type: 'com.linkedin.voyager.organization.OrganizationAddress',
+                headquarter: true,
+                line1: '1234 Lucky St',
+              },
+            ],
+          },
+        },
+        {
+          id: 'urn:book:1',
+          type: 'com.bookstore.ProjectedBook',
+          locations: [
+            {
+              country: 'US',
+              geographicArea: 'California',
+              city: 'San Francisco',
+              postalCode: '94110',
+              description: 'Club house',
+              $type: 'com.linkedin.voyager.organization.OrganizationAddress',
+              headquarter: true,
+              line1: '1234 Lucky St',
+            },
+          ],
+        },
+      ],
+    });
+
+    let model = this.store.peekRecord('com.bookstore.ProjectedBook', 'urn:book:1');
+
+    const currentCollection = model.get('locations').slice();
+    const aNewLocation = {
+      country: 'MX',
+      geographicArea: 'California',
+      city: 'Ensenada',
+      postalCode: '22810',
+      description: 'Home',
+      $type: 'com.linkedin.voyager.organization.OrganizationAddress',
+      headquarter: true,
+      line1: '555 Main St.',
+    };
+
+    model.set('locations', currentCollection.concat(aNewLocation));
+    // Saving relies on correctly setup nested record datas
+    await model.save();
```

Comment:
shouldn't we assert something after the save?

Reply:
We are making sure calling save doesn't error, I can put an assert in the mocked adapter to make sure it's called